packages/@dcos/sorting/README.md

```diff
@@ -0,0 +1,59 @@
+# DC/OS Sorting
+
+This project provides elementar sorting building blocks, which we use in the DC/OS UI.
+
+## Example
+
+```ts
+import {
+  sort,
+  reverseComparator,
+  compareAsNumber,
+  compareAsString
+} from "@dcos/sorting";
+
+function sortMyTable(data: Node[], isReversed) {
+  function getHostname(node: Node) {
+    if (!node.host) {
+      return null;
+    }
+
+    return node.host.name;
+  }
+
+  function getCpuUsage(node: Node) {
+    return node.cpuUsage;
+  }
+
+  return sort<Node>(
+    [compareAsString(getHostname), compareAsNumber(getCpuUsage)].map(
+      comparator => reverseComparator(comparator, isReversed)
+    ),
+    data
+  );
+}
+```
+
+## Combinators
+
+### `sort(comparatorsArray)`
+
+Sorts by the first comparator. If it is neutral (returns 0), sorts by the second. If it is neutral, ...
+
+### `reverseComparator(comparatorFunction, shouldBeReversed)`
+
+Takes a comparator and returns a new comparator that does the opposite
+
+## Comparators
+
+### `compareAsString(attributeLense)`
+
+Returns a function that uses the attributeLense to get the values to be compared as strings
+
+### `compareAsNumber(attributeLense)`
+
+Returns a function that uses the attributeLense to get the values to be compared as number
+
+## When to add a new comparator
+
+If you have a comparator which you want to use in DC/OS UI please make sure to add it here if it resembles a fundamental type. Examples for this would be number, string, health status, IP. Non-examples would be a node, a task or a service.
```

Comment:
Have you considered drawing this line a little lower? Like base types being only those from JS, and IP, Health and everything domain specific on DCOS being at the same place as Task, Service and etc...

Reply:
@Fabs As we are moving away from structs I don't really see a point in adding the comparators there. What would be the benefit? 

@nLight I don't really understand, can you explain this a bit more? Do you think we should add a section of when to add something for DC/OS UI in general?

Reply:
@DanielMSchmidt How a developers would know they should add a comparator in a sorting lib if they don't know about its existence? I mean we probably should add a line about the lib into https://github.com/dcos/dcos-ui/blob/master/CONTRIBUTING.md

Reply:
Ah, get you now 👍 will drop in a line

Reply:
@DanielMSchmidt 👍